lms/templates/widgets/footer-language-selector.html

```diff
@@ -0,0 +1,55 @@
+## Language-selection widget for the footer.
+##
+## Requires settings.LANGUAGE_COOKIE.
+<%page expression_filter="h"/>
+<%!
+  from babel import Locale
+  from django.conf import settings
+  from django.utils.translation import ugettext as _
+
+  from openedx.core.djangoapps.lang_pref import COOKIE_DURATION
+  from openedx.core.djangoapps.lang_pref.api import released_languages
+  from openedx.core.djangolib.js_utils import js_escaped_string
+
+  # Make sure LANGUAGE_COOKIE is present.
+  if not settings.LANGUAGE_COOKIE:
+      raise ValueError('settings.LANGUAGE_COOKIE is required to use footer-language-selector.')
+%>
+
+<div>
+    <label for="footer-language-selector">
+        <i class="fa fa-globe" aria-hidden="true"></i><span class="sr">${_("Choose Language")}</span>
+    </label>
+    <select id="footer-language-selector" onchange="setLanguage(this)">
+        % for language in sorted(released_languages(), key=lambda x: x.code):
+            <% language_name = Locale.parse(language.code.replace('_', '-'), sep='-').language_name %>
+            % if language.code == LANGUAGE_CODE:
+                <option value="${language.code}" selected="selected">${language_name}</option>
+            % else:
+                <option value="${language.code}">${language_name}</option>
+            % endif
+        % endfor
+    </select>
+</div>
+
+<script type="text/javascript">
+  ##
+  ## Set the language cookie using the same settings as
+  ## https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/lang_pref/views.py#L26
+  ## and
+  ## https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/lang_pref/middleware.py#63
+  ##
+  function setLanguage($select) {
+      var cookie = '${settings.LANGUAGE_COOKIE | n, js_escaped_string}=' + $select.value + ';path=/';
+      % if settings.SESSION_COOKIE_DOMAIN:
+          cookie += ';domain=${settings.SESSION_COOKIE_DOMAIN | n, js_escaped_string}';
+      % endif
+      % if COOKIE_DURATION:
+          cookie += ';max-age=${COOKIE_DURATION | n, js_escaped_string}';
+      % endif
+
+      document.cookie = cookie;
+      window.location.reload();
+  }
+</script>
```

Comment:
I'm wondering if you could post the request to the page and handle it on the server, the same way that update_lang does. You could pass the current location as a URL to redirect back to. Then you wouldn't need to have any JavaScript. I don't know whether this will do what we want in all the environments that show the footer, though.

What happens for a user who already has a language preference? Is this a temporary override in this browser only, or should it update their preference? I'm not sure that the UI makes this clear. Do you know how this is handled in the LMS selector in the header?